### DISCUSSION TITLE
Need Help: ESP32-WROOM + NMV2 1.6.3 Bitcoin Mining - Random Restarts Every 3-12 Hours (Session Best Difficulty/Uptime Reset to 0)

### DISCUSSION THREAD

**Discussion by surached3** [BitMaker-hub/NerdMiner_v2#512] [Q&A]
Hi everyone, I'm having an issue with Bitcoin mining on ESP32-WROOM and need help from experienced users.

Issue:
- Session Best Difficulty keeps resetting to 0 (current stats show 0.02)
- Uptime also resets to 0 at the same time (currently showing 31 minutes)
- From observation, these resets occur when Uptime is between 3-12 hours
- Seems like the board is auto-restarting

Current Mining Stats:
- Hash Rate: 59.0 KH/s
- Session Best Difficulty: 0.02
- Network Difficulty: 95.67T
- Session ID: ae6abf79

Setup:
- Board: ESP32-WROOM
- Power Supply: 5V 2A
- Software: NicehashMiner v2 (NMV2) version 1.6.3
- Connection: WiFi

Questions:
1. Is anyone else experiencing similar restart issues with NMV2 1.6.3 on ESP32-WROOM between 3-12 hours?
2. Power supply (5V 2A) should be sufficient, but still getting restarts - could it be Memory leak, Watchdog, or Overheat issues?
3. Are there any solutions to extend the running time without restarts?
4. Are there any special settings in NMV2 1.6.3 that might help reduce these restarts?

![{443888A5-734A-4D43-8229-F93137DCEA63}](https://github.com/user-attachments/assets/c9eac723-f25d-4c53-b404-2dcc9d5fb891)

Any help or suggestions would be greatly appreciated! 🙏

**Comment by adisorin** [comment]
Yes. I have the same problem with my ESP32 Wroom. I have two and one does the same as yours. I haven't found the cause. It is powered by 5V/2A and is in the room where the router is.

**Reply by ksevin** [reply]
Please let me know if you heard about anybody found a block with nerdminer?

**Reply by m8ax** [reply]
> Please let me know if you heard about anybody found a block with nerdminer?  

Nobody never

**Reply by m8ax** [reply]
> Yes. I have the same problem with my ESP32 Wroom. I have two and one does the same as yours. I haven't found the cause. It is powered by 5V/2A and is in the room where the router is.

What problem?

**Comment by ksevin** [comment]


**Reply by m8ax** [reply]
proobe my firmware... https://m8ax-nerdminerv2-firmware.netlify.app/

**Reply by m8ax** [reply]
for me no reboots with my firmware. here https://m8ax-nerdminerv2-firmware.netlify.app/ we have 3 firmwares one for nerdminerv2 esp32s3... see the pictures in web... other the 2.8 screen nerdminerv2 and the other is for esp32 wroom... the first 250kh/s the seconds 350.. and the third 395 without screen.. probe and comment.

**Comment by ksevin** [comment]


**Reply by m8ax** [reply]
in my firm ?

**Reply by m8ax** [reply]
what do yo use? for nerdmdinerv2 for 2.8 screen or what?

**Reply by m8ax** [reply]
probe my firmwares... Anyway, that has nothing to do with why it's not mining... I’ve set different timezones and it mines just fine. I told you yesterday to try my firmware."

**Comment by ksevin** [comment]


**Reply by m8ax** [reply]
probe my firmware[ https://m8ax-nerdminerv2-firmware.netlify.app

**Reply by m8ax** [reply]
xD

**Reply by m8ax** [reply]
what timezone hash you? where aure you from? sayme the zone notexactly its for see if i make a error in my firmware

**Reply by m8ax** [reply]
more later i make changes for change the hour automaticly in winter summer by ip... by the first configuration of miner must be correct the timezone... after ... alll automaticly,,,

**Reply by m8ax** [reply]
can you probe this firmware for any horus and sayme if the hour its correct in screens?
[firmware.zip](https://github.com/user-attachments/files/19615054/firmware.zip)

decompress the zip file and flash.. and sayme if the hour in screen its correct... and if pass about 3 hours the time its still corect... in this the change hour is automaticly by ip.. but the first configuration must be correct...

**Reply by m8ax** [reply]
in about 10 hours i upload in the web i say you after a new version with all corrected about time... and mega final version... enjoy it

**Reply by ksevin** [reply]
When I searched google about nerdminer, I have never met your page.   You need some rating😀.

**Reply by ksevin** [reply]
In a page about time zones and nerdminer I have read that UTC setting should be programmed in minutes because there are countries where utc is some hours and 30 minutes.  So the setting may be programmed as minutes.

**Reply by m8ax** [reply]
yes but its not programmedd to change the hour automaticly... in my firmware this knight when upload all correct exceot zones with +5:30  all times exact -1 -2- 3- 4- +4 +6+7+8+ etc... al lcorrect... and adjust automatic in summer and winter.. but remember the first configuration in web configuratiin must be correct..

**Reply by m8ax** [reply]
you have all fixed in my web ... install and say valoration final of al lthe firmware, thx. https://m8ax.github.io/MvIiIaX-NerdMiner_V2-DeV/ if you like yo can make a donation... very hard work..

**Reply by m8ax** [reply]
the temp is anecdotic this cpu not have sensor temop real... dont worry...  i have making a firmware for nmaxe too nmaxe an nmaxe gamma... and my donation ? xD

**Reply by ksevin** [reply]
Nothing so far except the speed.  The uptime has reset after 10 hours.  So I am waiting for a new psu to come.

**Comment by m8ax** [comment]
the hasrate must be 350. the latest actualization only change check timezone by ip.. this not more slowly the dispositive...  reboots sometimes are normal... i dont believe that you charger have a problem and the 32 bit shares its normal.. sometimes make more other not... its completly normal

**Comment by m8ax** [comment]
you say The uptime has reset after 10 hours. So I am waiting for a new psu to come.: whats the problem... its only takes a few secons to start mine... or prefer mine constantly at 56

**Comment by m8ax** [comment]
It has nothing to do with being online longer or more connected — absolutely nothing to do with it. f you like my firmware that much, you could consider making a donation... doing this kind of work takes effort, and not getting anything in return is honestly a bit disheartening... you can donate here: https://m8ax.github.io/MvIiIaX-NerdMiner_V2-DeV/ 😄... no harm in asking, right?

**Reply by m8ax** [reply]
or yo can considere suscribe to my yotube https://youtube.com/m8ax you have go this channel to the moon of suscribers. !! xD

**Reply by m8ax** [reply]
thank you... xD Anyway, don't worry if it ever restarts—it has nothing to do with difficulty, neither higher nor lower... But I prefer that it restarts occasionally but mines at 350, rather than never restarting and mining at 56. After all, how long does a restart take? 5 seconds? Anyway, it shouldn’t restart—mine has been running tests since yesterday without a single restart."

**Reply by m8ax** [reply]
On the other hand, I’ve noticed that the firmware website only displays properly at 1920x1080 resolution. At other resolutions, it looks a bit off... but oh well, everything is still understandable. If you want, you could test the website at different resolutions higher than Full HD and let me know how it looks. I’m not really sure how to make the page go viral so people try out my firmware… if you come up with any ideas, go ahead.

**Reply by m8ax** [reply]
in about 15hours i upoad a new version more quickly in menus and not down rate in certainly screens... i fixit now very things... in iphone this web not necesary to view ok? in iphon cant conect the nerdminer.....

**Comment by m8ax** [comment]
now yo can enter in the web https://m8ax.github.io/MvIiIaX-NerdMiner_V2-DeV/, i make other version. version with trump logo... all the versions for nerdminer of 2.8 screen now not ralentize and the functions are much better... in all screens 350 kh/s probe and sayme your opinion

**Comment by Wassim-Dib** [comment]
### Hola a todos
soy nuevo en esto de la mineria, el caso esque he flasheado un esp32 con NMV2 1.6.3, y no consigo avanzar mas alla del wifi que me pide password y el "MineYourCoins" da error. me podeis ayudar? Cual es el password del wifi?

**Comment by Wassim-Dib** [comment]
Hola a todos
soy nuevo en esto de la mineria, el caso esque he flasheado un esp32 con NMV2 1.6.3, y no consigo avanzar mas alla del wifi que me pide password y el "MineYourCoins" da error. me podeis ayudar? Cual es el password del wifi?

**Reply by m8ax** [reply]
Probe this firmware https://m8ax.github.io/MvIiIaX-NerdMiner_V2-DeV/
